src/components/ScanQRModal/index.tsx

```diff
@@ -28,22 +37,30 @@ const ScanQRModal = ({ classes, isOpen, onClose, onScan }) => {
   useEffect(() => {
     checkWebcam(
       () => {
-        setHasWebcam(true)
+        setUseWebcam(true)
       },
       () => {
-        setHasWebcam(false)
+        setUseWebcam(false)
       },
     )
   }, [])
 
   useEffect(() => {
-    // this fires only when the hasWebcam changes to false (null > false (user doesn't have webcam)
-    // , true > false (user switched from webcam to file upload))
-    // Doesn't fire on re-render
-    if (hasWebcam === false) {
+    if (useWebcam === false && !fileUploadModalOpen) {
+      setFileUploadModalOpen(true)
       openImageDialog()
     }
-  }, [hasWebcam, openImageDialog])
+  }, [useWebcam, openImageDialog, fileUploadModalOpen, setFileUploadModalOpen])
+
+  const onFileUploadHandlerClose = (successData, error) => {
```

Comment:
WDYT about error-first callbacks? It's quite a standard nowadays.

That way, `sucessData` can even be optional.

Also, can you add types to this function?